### PR TITLE
[TEVA-1516] Jobseekers can manage saved jobs

### PR DESCRIPTION
**Pull request by cpjmcquillan**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1516

## Changes in this PR
- Add saved jobs table
- Allow jobseekers to delete saved jobs

## Next steps
- Allow jobseekers to sort saved jobs

## Screenshots of UI changes
😃 
image
😢 
image

### CODE REVIEW

File: app/views/jobseekers/saved_jobs/index.html.haml [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
-= render partial: 'account_header'
+= render "account_header"
 
 .govuk-width-container
   .govuk-grid-row
-    .govuk-grid-column-two-thirds
+    .govuk-grid-column-full
       %h1.govuk-heading-l= t(".page_title")
+
+      - if @saved_jobs.present?
```
Comment (cesidio) [inline app/views/jobseekers/saved_jobs/index.html.haml:10]: `@saved_jobs.any?` would read better

File: spec/system/jobseekers_can_manage_their_saved_jobs_spec.rb [hunk 2]
```diff
@@ -10,12 +10,64 @@
...
+      let(:vacancy_1) { create(:vacancy) }
+      let(:vacancy_2) { create(:vacancy) }
+      let(:expired_vacancy) { create(:vacancy, :expired) }
+
+      before do
+        vacancy_1.organisation_vacancies.create(organisation: school)
+        vacancy_2.organisation_vacancies.create(organisation: school)
+        expired_vacancy.organisation_vacancies.create(organisation: school)
+
+        SavedJob.find_or_create_by(jobseeker_id: jobseeker.id, vacancy_id: vacancy_1.id)
```
Comment (cesidio) [inline spec/system/jobseekers_can_manage_their_saved_jobs_spec.rb:26]: to be consistent with what you just did on line 22 you could do:

`jobseeker.saved_jobs.create(vacancy: vacancy_1)`

do not think you need to do `find_or_create_by` in a spec

**Review by cesidio — APPROVED** [review]

File: spec/system/jobseekers_can_manage_their_saved_jobs_spec.rb [hunk 3]
```diff
@@ -10,12 +10,64 @@
...
+        expired_vacancy.organisation_vacancies.create(organisation: school)
+
+        jobseeker.saved_jobs.create(vacancy: vacancy_1)
+        jobseeker.saved_jobs.create(vacancy: vacancy_2)
+        jobseeker.saved_jobs.create(vacancy: expired_vacancy)
+
+        visit jobseekers_saved_jobs_path
+      end
+
+      context "and they view saved jobs" do
```
Comment (cesidio) [inline spec/system/jobseekers_can_manage_their_saved_jobs_spec.rb:33]: Last small thing, context should always start with `when` or `with`

Comment (cpjmcquillan) [reply spec/system/jobseekers_can_manage_their_saved_jobs_spec.rb:33]: Yes I agree, thanks

**Pull request merged**